### PR TITLE
docs: Explain how to update the changelog

### PR DESCRIPTION
**Pull request by phansch**
cc @Manishearth Is this more or less how you did it last time?

[Rendered](https://github.com/phansch/rust-clippy/blob/changelog_update/doc/changelog_update.md#changelog-update)

changelog: none

### CODE REVIEW

**Comment by phansch** [comment]
Image for the text Selection_064

File: doc/changelog_update.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+You'll want to run `util/fetch_prs_between.sh commit1 commit2 > changes.txt`
+and open that file in your editor of choice.
+
+* `commit1` is the Clippy commit hash of the previous stable release
+* `commit2` is the Clippy commit hash of the release you want to write the changelog for.
+
+When updating the changelog it's also a good idea to make sure that `commit1` is
+already correct in the current changelog.
+
+### 3. Authoring the final changelog
```
Comment (phansch) [inline doc/changelog_update.md]: This section should also mention updating the commit hash links

File: doc/changelog_update.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
+# Changelog Update
+
+If you want to help with updating the [changelog][changelog], you're in the right place.
+
+## When to update
+
+The changelog is ideally updated during the week before an upcoming stable
+release. Typos and other small fixes/additions are always welcome. You can find
```
Comment (phansch) [inline doc/changelog_update.md]: `Typos and other small fixes/additions are always welcome.` should be at the beginning of this paragraph and emphasized to encourage potential typo fixes.

File: doc/changelog_update.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+### 3. Authoring the final changelog
+
+The above script should have dumped all the relevant PRs to the file you
+specified. It should have filtered out most of the irrelevant PRs
+already, but it's a good idea to do a manual cleanup pass where you look for
+more irrelevant PRs. If you're not sure about some PRs, just leave them in for
+the review and ask for feedback.
+
+With PRs filtered, you can start to take each PR and move the
```
Comment (phansch) [inline doc/changelog_update.md]: `With the PRs filtered`

File: doc/changelog_update.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+### 1. Finding the relevant Clippy commits
+
+Each Rust release ships with its own version of Clippy. The Clippy submodule can
+be found in the [tools][tools] directory of the Rust repository.
+
+To find the Clippy commit hash for a specific Rust release you select the Rust
+release tag from the dropdown and then check the commit of the Clippy directory:
+
+![Explanation of how to find the commit hash](https://user-images.githubusercontent.com/2042399/62846160-1f8b0480-bcce-11e9-9da8-7964ca034e7a.png)
+
```
[the remark this answers is not in the document]
Comment (phansch) [reply doc/changelog_update.md]: Oh right, that's another option. I'm going to try that from scratch later today and add those instructions, too.

[the remark this answers is not in the document]
Comment (phansch) [reply doc/changelog_update.md]: @matthiaskrgr I've been doing:

```shell
git clone https://github.com/rust-lang/rust.git /tmp/rust && cd /tmp/rust
git checkout 1.34.0
git submodule update src/tools/clippy # first time needs --init, but always need to update it after checking out a new tag
cd src/tools/clippy
git log --format=%h -1
```
Is there an easier way?

[the remark this answers is not in the document]
Comment (matthiaskrgr) [reply doc/changelog_update.md]: Ah right, I forgot that clippy is not enabled in `config.toml` by default :( 
The way I proposed requires having clippy submodule enabled and some other parameters to be turned on that are not defaults... :/
I'm not sure anymore if cli-way is worth mentioning given then, what do you think?

[the remark this answers is not in the document]
Comment (phansch) [reply doc/changelog_update.md]: >  I'm not sure anymore if cli-way is worth mentioning given then, what do you think?

Yeah, I agree in this case :+1:

File: doc/changelog_update.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+## How to update
+
+### 1. Finding the relevant Clippy commits
+
+Each Rust release ships with its own version of Clippy. The Clippy submodule can
+be found in the [tools][tools] directory of the Rust repository.
+
+To find the Clippy commit hash for a specific Rust release you select the Rust
+release tag from the dropdown and then check the commit of the Clippy directory:
+
```
Comment (phansch) [inline doc/changelog_update.md]: Since I was confused just now.. This should also explain how to find the Clippy commit up the upcoming stable release.

**Comment by flip1995** [comment]
@phansch Since you most recently updated the changelog, you're the best one to judge, whether the documentation is comprehensive. If you think it is, r=me

**Comment by phansch** [comment]
> @phansch Since you most recently updated the changelog, you're the best one to judge, whether the documentation is comprehensive. If you think it is, r=me

I'll probably come up with more improvements the next time around, but I think this is good enough to get other people started, too.

@bors r=flip1995

**Pull request merged**